### PR TITLE
Registry perf2

### PR DESCRIPTION
**Pull request by ludoo0d0a**
push new clean branch for PR (based on 1.6.0 properly)

### CODE REVIEW

**Comment by S-Man42** [comment]
That looks better, but why is the Exif code in here. Was this intended?

Please close the PRs that are not needed anymore :)

File: lib/logic/tools/coords/data/coordinates.dart [hunk 1]
```diff
@@ -98,6 +98,18 @@ String _getSignString(int sign, bool isLatitude) {
   return _sign;
 }
 
+int getSignFromString(String text, bool isLatitude) {
+  int _sign = 0;
+
+  if (isLatitude) {
+    _sign = (text == 'N') ? 1 : -1;
+  } else {
+    _sign = (text == 'S') ? 1 : -1;
```
Comment (S-Man42) [inline lib/logic/tools/coords/data/coordinates.dart:107]: "S" needs to be "E" here.

File: lib/logic/tools/coords/data/coordinates.dart [hunk 2]
```diff
@@ -98,6 +98,18 @@ String _getSignString(int sign, bool isLatitude) {
   return _sign;
 }
 
+int getSignFromString(String text, bool isLatitude) {
```
Comment (S-Man42) [inline lib/logic/tools/coords/data/coordinates.dart:101]: `getCoordinateSignFromString` would be a better name. Because this is a public method, it is necessary to create a name which exactly points to the context. Otherwise it could happen that someone uses it for the wrong task ("sign string" could be used in several contextes)

File: lib/logic/tools/crypto_and_encodings/exif_reader.dart [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    LatLng _point = decToLatLon(DEC(_lat, _lng));
+
+    print("_point = ${_point}");
+    return _point;
+  }
+
+  return null;
+}
+
+double getCoordDec(IfdTag tag, String latlngRef, bool isLatitude) {
```
Comment (S-Man42) [inline lib/logic/tools/crypto_and_encodings/exif_reader.dart:62]: You always created public methods. Please make them private if they are not used anywhere else to keep the number of publicly usable methods as small as possible.

File: lib/logic/tools/crypto_and_encodings/exif_reader.dart [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+int getExifRotation(Map<String, IfdTag> tags) {
+  IfdTag orientation = tags["Image Orientation"];
+  int orientationValue = orientation.values[0];
+  int rotationCorrection = 0;
+  // in degress
+  print("orientation: ${orientation.printable}/${orientation.values[0]}");
+  switch (orientationValue) {
+    case 6:
```
Comment (S-Man42) [inline lib/logic/tools/crypto_and_encodings/exif_reader.dart:138]: what's with the other values?

File: lib/widgets/registry.dart [hunk 5]
```diff
@@ -708,21 +709,33 @@ class Registry {
...
+            SEARCHSTRING_FR_FORMULASOLVER,
           ]),
+
       GCWTool(tool: FormulaSolverFormulaGroups(), i18nPrefix: 'formulasolver', missingHelpLocales: [], searchStrings: [
         SEARCHSTRING_COMMON_FORMULASOLVER,
         SEARCHSTRING_DE_FORMULASOLVER,
         SEARCHSTRING_EN_FORMULASOLVER,
-        SEARCHSTRING_FR_FORMULASOLVER
-      ]),
+        SEARCHSTRING_FR_FORMULASOLVER,
```
Comment (S-Man42) [inline lib/widgets/registry.dart:731]: why did you add the commas?

File: lib/widgets/searchstrings/searchstrings_common.dart [hunk 6]
```diff
@@ -129,6 +129,7 @@ const SEARCHSTRING_COMMON_ESOTERIC_DEADFISH = 'deadfish idso xkcd ';
 const SEARCHSTRING_COMMON_ESOTERIC_MALBOLGE = 'malbolge olmstead';
 const SEARCHSTRING_COMMON_ESOTERIC_OOK = 'ook terry pratchett  ';
 const SEARCHSTRING_COMMON_ESOTERIC_WHITESPACELANGUAGE = 'whitespace';
+const SEARCHSTRING_COMMON_EXIF = 'exif';
```
Comment (S-Man42) [inline lib/widgets/searchstrings/searchstrings_common.dart:132]: other useful strings are
GERMAN: metadaten bilder
ENGLISH: metadata images pictures
FRENCH: ...

File: lib/widgets/registry.dart [hunk 7]
```diff
@@ -708,21 +709,33 @@ class Registry {
...
-            SEARCHSTRING_FR_ESOTERICPROGRAMMINGLANGUAGE
+            SEARCHSTRING_FR_ESOTERICPROGRAMMINGLANGUAGE,
+          ]),
+      GCWTool(
+          tool: ExifReader(),
+          i18nPrefix: 'exif',
+          category: ToolCategory.SCIENCE_AND_TECHNOLOGY,
+          missingHelpLocales: [],
+          searchStrings: [SEARCHSTRING_COMMON_EXIF,
+            SEARCHSTRING_DE_FORMULASOLVER,
```
Comment (S-Man42) [inline lib/widgets/registry.dart:722]: FORMULASOLVER searchstrings are wrong here :)

**Comment by S-Man42** [comment]
Please add tests for the logic. Especially the GPS Section would be very interesting since there are many edge cases to catch (-+90 als limits, so what happens if deg == 90 and minutes == 1? --> there's a normalize function in the coordinate sections... moreover, I'd tend to convert the coords into LatLon using the appropriate function, which already do such checks...; what havens if minutes >= 60, ...)

**Comment by ludoo0d0a** [comment]
I made really bad work .. i hate git branching
i need to clean up
exif has nothing to do here. bad cherry pick

File: lib/logic/tools/crypto_and_encodings/exif_reader.dart (hunk 4 above)
Comment (ludoo0d0a) [reply lib/logic/tools/crypto_and_encodings/exif_reader.dart:138]: no exif feature in this branch

File: lib/widgets/registry.dart (hunk 5 above)
Comment (ludoo0d0a) [reply lib/widgets/registry.dart]: no exif feature in this branch

File: lib/widgets/searchstrings/searchstrings_common.dart (hunk 6 above)
Comment (ludoo0d0a) [reply lib/widgets/searchstrings/searchstrings_common.dart]: no exif feature in this branch

File: lib/logic/tools/coords/data/coordinates.dart (hunk 1 above)
Comment (ludoo0d0a) [reply lib/logic/tools/coords/data/coordinates.dart:107]: no exif feature in this branch

File: lib/logic/tools/coords/data/coordinates.dart (hunk 2 above)
Comment (ludoo0d0a) [reply lib/logic/tools/coords/data/coordinates.dart:101]: no exif feature in this branch

File: lib/widgets/registry.dart (hunk 7 above)
Comment (ludoo0d0a) [reply lib/widgets/registry.dart]: no exif feature in this branch

File: lib/logic/tools/crypto_and_encodings/exif_reader.dart (hunk 3 above)
Comment (ludoo0d0a) [reply lib/logic/tools/crypto_and_encodings/exif_reader.dart:62]: no exif feature in this branch
